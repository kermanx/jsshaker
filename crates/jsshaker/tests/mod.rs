use std::fs;

use insta::{assert_snapshot, glob};
use jsshaker::{JsShakerOptions, TreeShakeConfig, tree_shake, vfs::SingleFileFs};
use oxc::{
  codegen::{CodegenOptions, CommentOptions},
  minifier::MinifierOptions,
};

fn do_tree_shake(input: String) -> String {
  let do_minify = input.contains("@minify");
  let react_jsx = input.contains("@react-jsx");
  let result = tree_shake(
    JsShakerOptions {
      vfs: SingleFileFs(input),
      config: TreeShakeConfig::recommended().with_react_jsx(react_jsx),
      minify_options: do_minify.then(|| MinifierOptions { mangle: None, ..Default::default() }),
      codegen_options: CodegenOptions { comments: CommentOptions::default(), ..Default::default() },
    },
    SingleFileFs::ENTRY_PATH.to_string(),
  );
  result.codegen_return[SingleFileFs::ENTRY_PATH].code.clone()
}

#[test]
fn test() {
  glob!("fixtures/**/*.js", |path| {
    let input = fs::read_to_string(path).unwrap();
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
      assert_snapshot!(do_tree_shake(input));
    })
  });
}
