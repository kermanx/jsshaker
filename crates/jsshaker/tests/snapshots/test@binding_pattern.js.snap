---
source: crates/jsshaker/tests/mod.rs
expression: do_tree_shake(input)
input_file: crates/jsshaker/tests/fixtures/binding_pattern.js
---
export function simple(unknown) {
	effect(1);
	effect(2);
	effect(3);
	effect(4);
	// Destructing unknown has effect
	let { g: { h, i: { j } } } = unknown;
}
export function with_rest(unknown) {
	let { ...rest } = { b: unknown };
	effect(void 0, rest.b, 2);
	let { b,...__unused_56F4 } = unknown;
	let { a: d } = { a: unknown };
	effect(d);
}
